docs/src/modules/utils/StyledEngineProvider.js

```diff
@@ -15,27 +15,20 @@ const cacheRtl = createCache({
 });
 
 export default function StyledEngineProvider(props) {
+  const { children, cacheLtr } = props;
   const theme = useTheme();
 
   const rtl = theme.direction === 'rtl';
-  let Wrapper = React.Fragment;
-  let wraperProps = {};
-
-  // Only happens client-side after the hydration
-  if (rtl) {
-    Wrapper = CacheProvider;
-    wraperProps = {
-      value: cacheRtl,
-    };
-  }
+  const emotionCache = theme.direction === 'rtl' ? cacheRtl : cacheLtr;
 
   return (
     <StyleSheetManager stylisPlugins={rtl ? [rtlPluginSc] : []}>
-      <Wrapper {...wraperProps}>{props.children}</Wrapper>
```

Comment:
Changing `Wrapper` meant that **all** `children` will get re-mounted. Especially high up in React trees, changing parent element types should be avoided at all cost.